common/src/main/java/dev/architectury/event/events/common/LootEvent.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * This file is part of architectury.
+ * Copyright (C) 2020, 2021, 2022 architectury
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package dev.architectury.event.events.common;
+
+import dev.architectury.event.Event;
+import dev.architectury.event.EventFactory;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.level.storage.loot.LootPool;
+import net.minecraft.world.level.storage.loot.LootTables;
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * Events related to loot tables and loot generation.
+ */
+public interface LootEvent {
+    /**
+     * An event to modify loot tables when they are loaded.
+     * This can be used to add new drops via new loot pools to existing loot tables
+     * without replacing the entire table.
+     *
+     * <h2>Example: adding diamonds as a drop for dirt</h2>
+     * <pre>{@code
+     * LootEvent.MODIFY_LOOT_TABLE.register((lootTables, id, context) -> {
+     *     // Check that the loot table is dirt
+     *     if (Blocks.DIRT.getLootTable().equals(id)) {
+     *         // Create a loot pool with a single item entry of Items.DIAMOND
+     *         LootPool.Builder pool = LootPool.lootPool().add(LootItem.lootTableItem(Items.DIAMOND));
+     *         context.addPool(pool);
+     *     }
+     * });
+     * }</pre>
+     *
+     * @see ModifyLootTable#modifyLootTable(LootTables, ResourceLocation, LootTableModificationContext)
+     */
+    Event<ModifyLootTable> MODIFY_LOOT_TABLE = EventFactory.createLoop();
+    
+    @FunctionalInterface
+    interface ModifyLootTable {
+        /**
+         * Modifies a loot table.
+         *
+         * @param lootTables the {@link LootTables} instance containing all loot tables
+         * @param id         the loot table ID
+         * @param context    the context used to modify the loot table
+         */
+        void modifyLootTable(LootTables lootTables, ResourceLocation id, LootTableModificationContext context);
```

Comment:
Would be nice as per discussed previously the `isBuiltin` param

Reply:
Yeah I've just forgotten about this pr, I'll fix it up today

Reply:
Okay done, also rebased onto master